File: benchmarks/fixtures/javascript-safe-preferences/src/preferences.js
const ALLOWED_PREFERENCES = new Set(["locale", "theme"]);

export function updatePreference(request, response, settings) {
  const key = String(request.body.key);
  if (!ALLOWED_PREFERENCES.has(key)) return response.status(400).end();
  settings.set(key, request.body.value);
  return response.json({ updated: key });
}

File: benchmarks/fixtures/python-safe-json/src/preferences.py
import json


def import_preferences(request):
    if len(request.body) > 4096:
        raise ValueError("Preference document is too large")
    payload = json.loads(request.body.decode("utf-8"))
    if not isinstance(payload, dict) or set(payload) - {"theme", "locale"}:
        raise ValueError("Unexpected preference fields")
    return {
        "theme": str(payload.get("theme", "system")),
        "locale": str(payload.get("locale", "en")),
    }
